### PR TITLE
Added list functionality without checkbox and X button

### PR DESCRIPTION
**Pull request by kdevsam**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -6,13 +6,15 @@
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
   <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
   <link rel="stylesheet" href="styles.css">
-
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
```
Comment (yazeedb) [inline public/index.html]: no jQuery

File: public/scripts/app.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ input.addEventListener('keypress', addTodo);
 function addTodo(e) {
    if (e.which !== 13) return;
    if (!e.target.value) return;
+    appendTodo(input);
+}
 
-   // do stuff
+function appendTodo(input){
+    $('ul').append('<li>' + input.value + '</li>');
```
Comment (yazeedb) [inline public/scripts/app.js]: naww
```js
document.querySelector('ul')
```

File: public/scripts/app.js [hunk 3]
```diff
@@ -4,6 +4,10 @@ input.addEventListener('keypress', addTodo);
 function addTodo(e) {
    if (e.which !== 13) return;
    if (!e.target.value) return;
+    appendTodo(input);
+}
 
-   // do stuff
+function appendTodo(input){
```
Comment (yazeedb) [inline public/scripts/app.js]: space the function out
```js
function appendTodo(input) {
  // do stuff
}
```

File: public/scripts/app.js [hunk 4]
```diff
@@ -4,6 +4,10 @@ input.addEventListener('keypress', addTodo);
...
+    $('ul').append('<li>' + input.value + '</li>');
+    input.value = "";
```
Comment (yazeedb) [inline public/scripts/app.js]: single quotes

File: public/styles.css [hunk 5]
```diff
@@ -25,7 +25,20 @@ input {
...
+ul{
+    width: 100%;
+    padding: 0;
+    margin: 0;
+}
+li{
+    display: list-item;
+    list-style-type: none;
+    padding: 16px;
+    font-size: 24px;
```
Comment (yazeedb) [inline public/styles.css]: maybe use em instead of px

File: public/index.html [hunk 6]
```diff
@@ -6,13 +6,15 @@
...
   <title>TodoMVC</title>
 </head>
 <body>
   <div class="container">
     <h1>todos</h1>
     <input type="text" placeholder="What needs to be done?">
+    <ul>
```
Comment (yazeedb) [inline public/index.html]: you could do `document.createElement('ul')`

File: public/index.html (hunk 1 above)
Comment (kdevsam) [inline public/index.html]: ah man lol

Comment (yazeedb) [inline public/index.html]: it has a lot of code we don't need. Regular DOM API + Vanilla JS should be all we need for this project 😄

File: public/scripts/app.js [hunk 7]
```diff
@@ -1,9 +1,21 @@
 const input = document.querySelector('input');
 input.addEventListener('keypress', addTodo);
+const container = document.createElement('ul');
+var counter = 0;
```
Comment (yazeedb) [inline public/scripts/app.js]: No `var`. If a variable's gonna be reassigned, use `let`.

The main difference is the scoping. `const`/`let` are block-scoped (Like C++ and Java), and `var` is function scoped. Example:

```js
function looper() {
  // i is the same variable
  for (var i = 0; i < 10; i++) {}
  for (var i = 0; i < 10; i++) {}

  // x is a different variable in both scopes
  for (let x = 0; x < 10; x++) {}
  for (let x = 0; x < 10; x++) {}
}
```

File: public/scripts/app.js [hunk 8]
```diff
@@ -1,9 +1,21 @@
...
 
 function addTodo(e) {
    if (e.which !== 13) return;
    if (!e.target.value) return;
+    appendTodo(input);
+}
 
-   // do stuff
+function appendTodo(input) {
+  counter++;
```
Comment (yazeedb) [inline public/scripts/app.js]: shouldn't need a counter

File: public/scripts/app.js [hunk 9]
```diff
@@ -1,9 +1,21 @@
...
+  var list = document.createElement('li');
+  t[counter] = document.createTextNode(input.value);
```
Comment (yazeedb) [inline public/scripts/app.js]: you could also do `list.innerHTML = input.value`

**Comment by yazeedb** [comment]
We should just need:

* a function to validate the input on submit (input event handler)
* a function to create a list item of that input
* a function to append that item to the todo list

File: public/styles.css [hunk 10]
```diff
@@ -25,7 +25,20 @@ input {
   border: none;
   outline: none;
 }
-
+ul{
+    width: 66%;
```
Comment (yazeedb) [inline public/styles.css]: the container's already 66%, this would make its width 43.56% 😨

File: public/scripts/app.js [hunk 11]
```diff
@@ -11,11 +10,11 @@ function addTodo(e) {
...
-  counter++;
-  var list = document.createElement('li');
-  t[counter] = document.createTextNode(input.value);
+  let list = document.createElement('li');
+list.innerHTML = input.value;
+  
   container.appendChild(list);
-  list.appendChild(t[counter]);
+  
   document.body.appendChild(container);
```
Comment (yazeedb) [inline public/scripts/app.js]: body will append the container every time this function's called

File: public/scripts/app.js [hunk 12]
```diff
@@ -11,11 +10,11 @@ function addTodo(e) {
 }
 
 function appendTodo(input) {
-  counter++;
-  var list = document.createElement('li');
-  t[counter] = document.createTextNode(input.value);
+  let list = document.createElement('li');
```
Comment (yazeedb) [inline public/scripts/app.js]: const

File: public/scripts/app.js [hunk 13]
```diff
@@ -11,11 +10,11 @@ function addTodo(e) {
...
+list.innerHTML = input.value;
+  
   container.appendChild(list);
-  list.appendChild(t[counter]);
+  
   document.body.appendChild(container);
-    input.value = '';
+  input.value = '';
```
Comment (yazeedb) [inline public/scripts/app.js]: no need to empty it out. And why `input.value`? Why not just pass the text right in?
Is the function expecting something like `{ value: 'todo item' }`?

Comment (kdevsam) [inline public/scripts/app.js]: you need to clear the text after pressing enter thats what that line of code does.

File: public/scripts/app.js (hunk 11 above)
Comment (kdevsam) [inline public/scripts/app.js]: ugh true - fixing rn 👍

File: public/scripts/app.js [hunk 14]
```diff
@@ -1,9 +1,17 @@
 const input = document.querySelector('input');
 input.addEventListener('keypress', addTodo);
+const container = document.createElement('ul');
+document.body.appendChild(container);
 
 function addTodo(e) {
    if (e.which !== 13) return;
    if (!e.target.value) return;
+    appendTodo(input);
```
Comment (yazeedb) [inline public/scripts/app.js]: Oh, ok. I see what you're doing. This is could cause "side effects" by giving `appendTodo` unnecessary access to the `input` element. Just pass `input.value` directly to `appendTodo`, and reset `input.value` inside of addTodo

File: public/scripts/app.js [hunk 15]
```diff
@@ -1,9 +1,17 @@
...
+}
 
-   // do stuff
+function appendTodo(input) {
```
Comment (yazeedb) [inline public/scripts/app.js]: call this something like `createTodoItem` and have it just return an `<li>` with the text inside of it. Then inside `appendTodo`, append that list item inside of `container`

**Pull request merged**